packages/components/src/components/Flyout/Flyout.tsx

```diff
@@ -0,0 +1,172 @@
+    /**
+     * Define the prefered alignment of the flyout. Can be overriden by viewport collisions viewport.
+     * @default start
+     */
+    align?: 'start' | 'center' | 'end';
+    /**
+     * Define the padding of the flyout
+     * @default compact
+     */
+    padding?: 'compact' | 'comfortable' | 'spacious';
+    /**
+     * Define the maximum width of the flyout
+     * @default 360px
+     */
+    maxWidth?: string;
+    children?: ReactNode;
+    'data-test-id'?: string;
+};
+
+export type FlyoutTriggerProps = { children?: ReactNode; 'data-test-id'?: string };
+
+export type FlyoutHeaderProps = {
+    /**
+     * Show a close button in the header
+     * @default false
+     */
+    showCloseButton?: boolean;
+    children?: ReactNode;
+    'data-test-id'?: string;
+};
+
+export type FlyoutFooterProps = { children?: ReactNode; 'data-test-id'?: string };
+
+export type FlyoutBodyProps = { children?: ReactNode; 'data-test-id'?: string };
+
+export const FlyoutRoot = ({ children, ...props }: FlyoutRootProps) => {
+    return <RadixPopover.Root {...props}>{children}</RadixPopover.Root>;
+};
+FlyoutRoot.displayName = 'Flyout.Root';
+
+export const FlyoutTrigger = (
+    { children, 'data-test-id': dataTestId = 'fondue-flyout-trigger' }: FlyoutTriggerProps,
+    ref: ForwardedRef<HTMLButtonElement>,
+) => {
+    return (
+        <RadixPopover.Trigger data-test-id={dataTestId} asChild ref={ref}>
+            {children}
+        </RadixPopover.Trigger>
+    );
+};
+FlyoutTrigger.displayName = 'Flyout.Trigger';
+
+export const FlyoutContent = (
+    {
+        align = 'start',
+        maxWidth = '360px',
+        padding = 'compact',
+        'data-test-id': dataTestId = 'fondue-flyout-content',
+        children,
+        ...props
+    }: FlyoutContentProps,
+    ref: ForwardedRef<HTMLDivElement>,
+) => {
+    return (
+        <RadixPopover.Portal>
+            <RadixPopover.Content
+                style={
+                    {
+                        '--flyout-max-width': maxWidth,
+                    } as CSSProperties
+                }
+                ref={ref}
+                align={align}
+                collisionPadding={8}
+                sideOffset={8}
+                className={flyoutContentStyles({ ...props })}
+                data-flyout-spacing={padding}
+                data-test-id={dataTestId}
+                {...props}
+            >
+                {children}
+            </RadixPopover.Content>
+        </RadixPopover.Portal>
+    );
+};
+FlyoutContent.displayName = 'Flyout.Content';
+
+export const FlyoutHeader = (
+    { showCloseButton, children, 'data-test-id': dataTestId = 'fondue-flyout-header' }: FlyoutHeaderProps,
+    ref: ForwardedRef<HTMLDivElement>,
+) => {
+    return (
+        <div data-test-id={dataTestId} ref={ref} className={flyoutHeaderStyles}>
+            <div>{children}</div>
+            {showCloseButton && (
+                <RadixPopover.Close className="tw-cursor-pointer">
```

Comment:
Could you use the role `button` or add a test id on `RadixPopover.Close` and use it in the test instead of the Fondue icon test id.

Reply:
adjusted